kibicara/kibicara_mda.py

```diff
@@ -0,0 +1,52 @@
+# Copyright (C) 2020 by Maike <maike@systemli.org>
+#
+# SPDX-License-Identifier: 0BSD
+
+import email.parser
+import sys
+import re
+import requests
+from logging import getLogger
+from kibicara.model import Hood
+from kibicara.platforms.email.webapi import get_email
+import argparse
+
+
+async def main():
+    logger = getLogger(__name__)
+
+    # the MDA passes the recipient address as command line argument
+    parser = argparse.ArgumentParser()
+    parser.add_argument("recipient_address")
+    args = parser.parse_args()
+
+    # read mail from STDIN
+    mailbytes = bytes(sys.stdin.read())
+
+    # parse plaintext to email.EmailMessage object
+    myparser = email.parser.BytesParser()
+    mail = myparser.parsebytes(mailbytes)
+
+    # extract relevant data from mail
+    for part in mail.walk():
+        try:
+            text = part.get_body(('plain',))
+            if not text:
+                text = re.sub(r'<[^>]*>', '', part.get_body(('html',)))
+        except Exception:
+            logger.info("No Body in this message part", exc_info=True)
+    if not text:
+        logger.error('No suitable message body')
+        exit(0)
+    # extract hood name from the envelope recipient address
+    hood_name = args.recipient_address.split('@')[0]
+    hood = await Hood.objects.get(name=hood_name)
+    email_row = await get_email(hood=hood.id)
```

Comment:
what does ETA mean in this context?

Yes, disabling the email bot should be possible in the long run; some admins may want this because of spam problems or so.

I'm not really a fan of a mandatory email address prefix... looks a bit ugly and is unintuitive for users.
What's the problem if someone registers a `root` hood? Emails probably wouldn't arrive at the bot, but at that point it's their problem.... I guess the MTA should sort out those mails before transmitting them to the MDA. An exhaustive list of special names would be good in case of MTA misconfiguration.

Reply:
Creating a hood does not create an email bot, so it is optional currently.
The problem is that if they can get emails `hostmaster`/..., they can get TLS certificates for the domain. MTA configuration requires the exhaustive list of names on the MTA side (and is a footgun for the siteadmin). Do you know which names are accepted as signifier of controlling the domain for every CA?

Reply:
Automatically creating the email bot or letting the admin decide is a UX question... it could be good to let the admin decide themselves, so they learn about the feature and recognize to promote it. OTOH maybe it's sufficient if it's in the template of the hood page.

Reply:
There is this list for reserved names: https://webmasters.stackexchange.com/questions/104811/is-there-any-list-of-email-addresses-reserved-because-of-security-concerns-for-a#105020

Reply:
I opened #40 and will implement a prefix for now; we can take that limitation out after #40 is addressed.